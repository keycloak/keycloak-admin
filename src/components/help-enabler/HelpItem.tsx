import React, { useContext } from "react";
import { Popover } from "@patternfly/react-core";
import { HelpIcon } from "@patternfly/react-icons";
import { useTranslation } from "react-i18next";
import { HelpContext } from "./HelpHeader";

type HelpItemProps = {
  helpText: string;
  forLabel: string;
  forID: string;
};

export const HelpItem = ({ helpText, forLabel, forID }: HelpItemProps) => {
  const { t } = useTranslation();
  const { enabled } = useContext(HelpContext);
  return (
    <>
      {enabled && (
        // <Tooltip position="right" content={t(`help:${item}`)}>
        //   <span id={item} data-testid={item}> *** I don't know what data-testid was being used for ***
        //     <HelpIcon />
        //   </span>
        // </Tooltip>
        // <Popover bodyContent={t(`help:${item}`)}>
        <Popover bodyContent={helpText}>
          <button
            aria-label={t(`helpLabel`, { label: forLabel })}
            onClick={(e) => e.preventDefault()}
            aria-describedby={forID}
            className="pf-c-form__group-label-help"
          >
            <HelpIcon noVerticalAlign />
          </button>
        </Popover>
      )}
    </>
  );
};
