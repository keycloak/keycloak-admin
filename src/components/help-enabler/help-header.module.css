
.helpDropdownItem {
  background-color: transparent;
}

.helpText {
  overflow-wrap: break-word;
  white-space: normal;
}