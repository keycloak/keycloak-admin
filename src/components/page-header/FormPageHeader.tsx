import React, { useContext } from "react";
import {
  PageSection,
  Text,
  TextContent,
  Divider,
} from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import { HelpContext } from "../help-enabler/HelpHeader";

type FormPageHeaderProps = {
  titleKey: string;
  subKey: string;
};

export const FormPageHeader = ({ titleKey, subKey }: FormPageHeaderProps) => {
  const { t } = useTranslation();
  const { enabled } = useContext(HelpContext);
  return (
    <>
      <PageSection variant="light">
        <TextContent>
          <Text component="h1">{t(titleKey)}</Text>
          {enabled && t(`help:${subKey}`)}
        </TextContent>
      </PageSection>
      <Divider />
    </>
  );
};
