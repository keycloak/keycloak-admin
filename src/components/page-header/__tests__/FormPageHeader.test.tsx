import React, { useContext } from "react";
import { Button } from "@patternfly/react-core";
import { render } from "@testing-library/react";

import { FormPageHeader } from "../FormPageHeader";
import { HelpContext, Help } from "../../help-enabler/HelpHeader";
import { mount } from "enzyme";

describe("<FormPageHeader />", () => {
  it("render", () => {
    const comp = render(<FormPageHeader titleKey="dummy" subKey="dummy" />);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it("with help disabled", () => {
    const ExampleFormPage = () => {
      const { toggleHelp } = useContext(HelpContext);
      return (
        <>
          <Button onClick={() => toggleHelp()}>Toggle</Button>
          <FormPageHeader
            titleKey="storybookTitle"
            subKey="storybookSubTitle"
          />
        </>
      );
    };

    const wrapper = mount(
      <Help>
        <ExampleFormPage />
      </Help>
    );
    const button = wrapper.find("button");
    expect(button).not.toBeNull();
    expect(wrapper.text()).toEqual("TogglestorybookTitlestorybookSubTitle");

    button.simulate("click");
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual("TogglestorybookTitle");
  });
});
