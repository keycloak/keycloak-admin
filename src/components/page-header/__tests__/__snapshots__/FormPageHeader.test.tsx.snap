// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormPageHeader /> render 1`] = `
<DocumentFragment>
  <section
    class="pf-c-page__main-section pf-m-light"
  >
    <div
      class="pf-c-content"
    >
      <h1
        class=""
        data-pf-content="true"
      >
        dummy
      </h1>
      dummy
    </div>
  </section>
  <hr
    class="pf-c-divider"
  />
</DocumentFragment>
`;

exports[`<FormPageHeader /> with help disabled 1`] = `
<Help>
  <ExampleFormPage>
    <Button
      onClick={[Function]}
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-primary"
        data-ouia-component-id={1}
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Toggle
      </button>
    </Button>
    <FormPageHeader
      subKey="storybookSubTitle"
      titleKey="storybookTitle"
    >
      <PageSection
        variant="light"
      >
        <section
          className="pf-c-page__main-section pf-m-light"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                component="h1"
              >
                <h1
                  className=""
                  data-pf-content={true}
                >
                  storybookTitle
                </h1>
              </Text>
            </div>
          </TextContent>
        </section>
      </PageSection>
      <Divider>
        <hr
          className="pf-c-divider"
        />
      </Divider>
    </FormPageHeader>
  </ExampleFormPage>
</Help>
`;
